src/core/config/npo/identifier.rs

```diff
@@ -56,33 +80,50 @@ impl<'a> Identifier<'a> {
     #[inline(always)]
     fn find_fan_out(&mut self, type_name: &'a str, is_list: bool) -> Queries<'a> {
```

Comment:
```suggestion
    fn create_index(&mut self, type_name: &'a str, is_list: bool) -> Queries<'a> {
```
- Make it non-recursive
- Return a `HashMap<TypeName, Vec<(FieldName, TypeName, isList: bool, isBatched: bool)>>`